src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -217,6 +244,39 @@ private IReadOnlyList<SemanticRange> CombineSemanticRanges(params IReadOnlyList<
             return result;
         }
 
+        private async Task<SemanticTokens> GetMatchingCSharpResponseAsync(
+            TextDocumentIdentifier textDocumentIdentifier,
+            long? documentVersion,
+            CancellationToken cancellationToken)
+        {
+            var parameter = new SemanticTokensParams
+            {
+                TextDocument = textDocumentIdentifier,
+            };
+
+            ProvideSemanticTokensResponse csharpResponse;
+            for (var i = 0; i < 3; i++)
```

Comment:
Is `3` a retry number? Would it help to say `const int maxRetries = 3; for (var i = 0; i < maxRetries; i++)`?